extensions/resteasy-reactive/jaxrs-client-reactive/deployment/src/main/java/io/quarkus/jaxrs/client/reactive/deployment/JaxrsClientReactiveProcessor.java

```diff
@@ -453,293 +441,376 @@ public void close() {
-                }
+                            String[] consumes = extractProducesConsumesValues(
+                                    jandexSubMethod.declaringClass().classAnnotation(CONSUMES), method.getConsumes());
+                            consumes = extractProducesConsumesValues(jandexSubMethod.annotation(CONSUMES), consumes);
+                            handleReturn(subResourceInterface, defaultMediaType,
+                                    getHttpMethod(jandexSubMethod, subMethod.getHttpMethod(), httpAnnotationToMethod),
+                                    consumes, jandexSubMethod, subMethodCreator, bodyParameterValue, builder);
+                        }
 
-                catchBlock.throwException(caughtException);
+                        if (subConstructor != null) {
+                            subConstructor.returnValue(null);
+                        }
 
-                if (bodyParameterIdx != null) {
-                    String[] consumes = method.getConsumes();
-                    if (consumes != null && consumes.length > 0) {
+                        methodCreator.returnValue(subInstance);
+                    }
+                } else {
 
-                        if (consumes.length > 1) {
-                            throw new IllegalArgumentException(
-                                    "Multiple `@Consumes` values used in a MicroProfile Rest Client: " +
-                                            restClientInterface.getClassName()
-                                            + " Unable to determine a single `Content-Type`.");
+                    // constructor: initializing the immutable part of the method-specific web target
+                    FieldDescriptor webTargetForMethod = FieldDescriptor.of(name, "target" + methodIndex, WebTarget.class);
+                    c.getFieldCreator(webTargetForMethod).setModifiers(Modifier.FINAL);
+                    webTargets.add(webTargetForMethod);
+
+                    AssignableResultHandle constructorTarget = createWebTargetForMethod(constructor, baseTarget, method);
+                    constructor.writeInstanceField(webTargetForMethod, constructor.getThis(), constructorTarget);
+
+                    // generate implementation for a method from jaxrs interface:
+                    MethodCreator methodCreator = c.getMethodCreator(method.getName(), method.getSimpleReturnType(),
+                            javaMethodParameters);
+
+                    AssignableResultHandle methodTarget = methodCreator.createVariable(WebTarget.class);
+                    methodCreator.assign(methodTarget,
+                            methodCreator.readInstanceField(webTargetForMethod, methodCreator.getThis()));
+
+                    Integer bodyParameterIdx = null;
+                    Map<MethodDescriptor, ResultHandle> invocationBuilderEnrichers = new HashMap<>();
+
+                    for (int paramIdx = 0; paramIdx < method.getParameters().length; ++paramIdx) {
```

Comment:
I don't really like that there are 3 parameter loops, but I don't know if it would just cause more complexity if we tried to extract this into a method.